From 77da3b36b68482d1d2309a0725da399196682de0 Mon Sep 17 00:00:00 2001
From: "Bernhard M. Wiedemann" <bwiedemann@suse.de>
Date: Mon, 5 Jun 2017 21:37:38 +0200
Subject: [PATCH 5/6] sort input files

when building packages (e.g. for openSUSE Linux)
(random) filesystem order of input files
influences ordering of entries in the output,
thus without the patch, builds (in disposable VMs) would usually differ.

See https://reproducible-builds.org/ for why this matters.
---
 python/volk_modtool/volk_modtool_generate.py | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

--- a/python/volk_modtool/volk_modtool_generate.py
+++ b/python/volk_modtool/volk_modtool_generate.py
@@ -58,10 +58,10 @@
         else:
             name = self.get_basename(base)
         if name == '':
-            hdr_files = glob.glob(os.path.join(base, "kernels/volk/*.h"))
+            hdr_files = sorted(glob.glob(os.path.join(base, "kernels/volk/*.h")))
             begins = re.compile("(?<=volk_).*")
         else:
-            hdr_files = glob.glob(os.path.join(base, "kernels/volk_" + name + "/*.h"))
+            hdr_files = sorted(glob.glob(os.path.join(base, "kernels/volk_" + name + "/*.h")))
             begins = re.compile("(?<=volk_" + name + "_).*")
 
         datatypes = []
@@ -156,7 +156,7 @@
         open(dest, 'w+').write(outstring)
 
         # copy orc proto-kernels if they exist
-        for orcfile in glob.glob(inpath + '/kernels/volk/asm/orc/' + top + name + '*.orc'):
+        for orcfile in sorted(glob.glob(inpath + '/kernels/volk/asm/orc/' + top + name + '*.orc')):
             if os.path.isfile(orcfile):
                 instring = open(orcfile, 'r').read()
                 outstring = re.sub(oldvolk, 'volk_' + self.my_dict['name'], instring)
